db/error_handler.cc

```diff
@@ -543,12 +543,20 @@ Status ErrorHandler::StartRecoverFromRetryableBGIOError(IOStatus io_error) {
   if (bg_error_.ok() || io_error.ok()) {
     return Status::OK();
   }
-  if (db_options_.max_bgerror_resume_count <= 0 || recovery_in_prog_ ||
-      recovery_thread_) {
+  if (db_options_.max_bgerror_resume_count <= 0 || recovery_in_prog_) {
     // Auto resume BG error is not enabled, directly return bg_error_.
     return bg_error_;
   }
 
+  if (recovery_thread_) {
```

Comment:
Would it make sense to reset ```recovery_thread_``` around line 470, so that way its guaranteed to be null when this is called for a new error?

Reply:
ClearBGError() is not used in the auto recovery thread and it is only used by the manual resume. If we resert recovery_thread_ in it, actually when next auto resume comes, the issue is still there.